File: problems/013-build-fingerprint/solutions/python/main.py
import sys

it = iter(sys.stdin.read().split())
n = int(next(it))
q = int(next(it))
f = [(next(it), next(it)) for _ in range(n)]
output = sys.stdout
for _ in range(q):
    meta = [next(it) for _ in range(4)]
    k = int(next(it))
    ids = [int(next(it)) - 1 for _ in range(k)]
    ids.sort(key=lambda x: f[x][0])
    output.write(meta[0])
    for token in (*meta[1:], str(k)):
        output.write(" ")
        output.write(token)
    for x in ids:
        output.write(" ")
        output.write(f[x][0])
        output.write(" ")
        output.write(f[x][1])
    output.write("\n")

File: problems/013-build-fingerprint/solutions/rust/main.rs
use std::io::{self, BufWriter, Read, Write};
fn main() {
    let mut s = String::new();
    io::stdin().read_to_string(&mut s).unwrap();
    let mut it = s.split_whitespace();
    let n: usize = it.next().unwrap().parse().unwrap();
    let q: usize = it.next().unwrap().parse().unwrap();
    let mut f = Vec::with_capacity(n);
    for _ in 0..n {
        f.push((
            it.next().unwrap().to_string(),
            it.next().unwrap().to_string(),
        ));
    }
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    for _ in 0..q {
        let meta: [&str; 4] = std::array::from_fn(|_| it.next().unwrap());
        let k: usize = it.next().unwrap().parse().unwrap();
        let mut ids: Vec<usize> = (0..k)
            .map(|_| it.next().unwrap().parse::<usize>().unwrap() - 1)
            .collect();
        ids.sort_by(|&a, &b| f[a].0.cmp(&f[b].0));
        write!(out, "{} {} {} {} {}", meta[0], meta[1], meta[2], meta[3], k).unwrap();
        for x in ids {
            write!(out, " {} {}", f[x].0, f[x].1).unwrap();
        }
        writeln!(out).unwrap();
    }
}
